open_optical_gating/cli/determine_reference_period.py

```diff
@@ -118,6 +155,12 @@ def estimate_period_length(diffs):
         1,
     ]  # list of values needed in the gotScoreForDelta function: minScore, maxScore, totalScore, meanScore, minSinceMax, deltaForMinSinceMax, stage, numScores
     # TODO WTF, can this be simplified?
+    # TODO: JT writes: I assume the above comment refers to the list of values. Ugh. I agree, this really needs tidying into a dictionary
+    # or, dare I say it, a class(!). This might be where your and my class philosophies do diverge a little; it's a borderline one for me even.
+    # But, I think it's a reasonable concept to have a class instance (lifetime could be while we are in the establish-period state...)
+    # whose job it is to determine the period. There is internal state to be maintained (as demonstrated by the passing-around of 'values'),
+    # and when running live (not just determining period in one go, from one big historical frame buffer) we might potentially want
+    # some of that state to persist from one received frame to the next. So there are benefits to maintaining, and encapsulating, that state in one object.
```

Comment:
Agreed that a dictionary or class is probably the way to go. Dictionaries are nice in terms because they are very happy to be stored in session data or passed around in requests once jsonified. I have argument against it being a custom object except this.

Reply:
Resolving this because, in this case, the solution is just to inline gotScoreForDelta. There's a TODO in the source code about this.